control_protocol.md

```diff
@@ -0,0 +1,296 @@
+A Coordinator shall follow the {ref}`sign out routine<sign-out>` for a signed in Component considered dead.
+
+:::{note}
+TBD: Respond to every non empty message with an empty one?
+:::
+
+
+(sign-out)=
+##### Signing out
+
+A Component should tell a Coordinator, when it stops working, with a SIGNOUT message.
+The Coordinator shall ACKNOWLEDGE the sign out and remove the ID from its address book.
+It shall also publish to the other Coordinators in the network that this Component signed out, see {ref}`Coordinator coordination<coordinator-coordination>`.
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> N1: COORDINATOR|N1.CA|SIGNOUT
+    participant N1 as N1.COORDINATOR
+    N1 ->> CA: N1.CA|N1.COORDINATOR|ACKNOWLEDGE
+    Note right of N1: Removes "CA" with identity "IA"<br> from address book
+    Note left of CA: Shall not send any message anymore except SIGNIN
+:::
+
+
+#### Communication with other Components
+
+The following two examples show, how a message is transferred between two components `CA`, `CB` via one or two Coordinators.
+Coordinators shall send messages from their DEALER socket to other Coordinator's ROUTER socket.
+
+Coordinators shall hand on the message to the corresponding Coordinator or connected Component.
+
+
+:::{mermaid}
+sequenceDiagram
+    alt full ID
+        CA ->> N1: N1.CB|N1.CA| Give me property A.
+    else only Component ID
+        CA ->> N1: CB|N1.CA| Give me property A.
+    end
+    participant N1 as N1.COORDINATOR
+    N1 ->> CB: N1.CB|N1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> N1: N1.CA|N1.CB| Property A has value 5.
+    N1 ->> CA: N1.CA|N1.CB| Property A has value 5.
+:::
+
+
+:::{mermaid}
+sequenceDiagram
+    CA ->> N1: N2.CB|N1.CA| Give me property A.
+    participant N1 as N1.COORDINATOR
+    Note over N1,N2: N1 DEALER socket sends to N2 ROUTER
+    participant N2 as N2.COORDINATOR
+    N1 ->> N2: N2.CB|N1.CA| Give me property A.
+    N2 ->> CB: N2.CB|N1.CA| Give me property A.
+    Note left of CB: Reads property A
+    CB ->> N2: N1.CA|N2.CB| Property A has value 5.
+    Note over N1,N2: N2 DEALER socket sends to N1 ROUTER
+    N2 ->> N1: N1.CA|N2.CB| Property A has value 5.
+    N1 ->> CA: N1.CA|N2.CB| Property A has value 5.
+:::
+
+Prerequisite of Communication between two Components are:
+- Both Components are connected to a Coordinator and {ref}`signed in<sign-in>`.
+- Both Components are either connected to the same Coordinator (example one), or their Coordinators are connected to each other (example two).
+
+
+The following flow chart shows the decision scheme and message modification in a Coordinator `Co1` of Node `N1`.
+Its full ID is `N1.Coordinator`.
+`n0`, `nR` are placeholders for sender and recipient Node ID.
+`recipient` is a placeholder for the recipient Component ID.
+`IA` is the connection identity of the Component `N1.CA` (if it is directly connected to `Co1`) or of its Coordinator `n0.COORDINATOR`.
```

Comment:
I think what is confusing me is the mix of generality and specificity. We have a Component `CA` on a Node `N1`, how can the name of its Coordinator be anything but `N1.COORDINATOR`? I think it shows through, as you felt you should clarify in the arrow texts that `|n0.CA|` == in fact `|N1.CA|` (which further clutters up the graph) :-/

Maybe it works better if it's all numbers/letters, or all "S" and "R" designators?

Reply:
Also, I get that `IA` is the connection identity of `N1.CA`, but how can it _alternatively_ be the identity of `N1.CA`'s ("its") Coordinator :interrobang:

Reply:
I tried to show the path of an arbitrary message through the system, depending on different cases.

It seems, not very successful.

> Also, I get that IA is the connection identity of N1.CA, but how can it alternatively be the identity of N1.CA's ("its") Coordinator ⁉️

Oh  yeah, that should be a variable as well .

Reply:
> Maybe it works better if it's all numbers/letters, or all "S" and "R" designators?

so, no specifics?

Reply:
I'd say, either all specific (fixed numbers/letters), or all generic (S/R), not a mixture.  :shrug:

Reply:
Ok. I'll try it.

Reply:
I think there's nothing more to be done here, resolving.